### PR TITLE
Update to use latest lite-core and enable version vector

### PR DESCRIPTION
**Pull request by pasin**
I have made two separate commits when updating Lite-Core: Before and After enabling version-vector. Here are summary of the changes:

**Before enabling version-vector:**
* Used `c4doc_getProperties` instead of `c4doc.selectedRev.body` and ensured to retain the body through the life of the document according to its selected revision. This has been done in CBLC4Document that wraps and retains C4Document object.
* Moved `-encode:` method from `CBLMutableDocument` to `CBLDocument` itself as overriding is not needed anymore.
* Used `c4db_getDoc` instead of `c4doc_get` as `c4doc_get` is semi-deprecated.
* Used `C4DocContentLevel.kDocGetAll` when creating a conflicting doc in `CBLDatabase`’s `-resolveConflictInDocument:` as we need to select a conflicting revision.
* Added another constructor in `CBLDocument` to be able to specify `C4DocContentLevel` when using `c4db_getDoc`.
* Disabled some tests per the open issues.
* Open issues: [CBL-1707](https://issues.couchbase.com/browse/CBL-1707), [CBL-1709](https://issues.couchbase.com/browse/CBL-1709), [CBL-1710](https://issues.couchbase.com/browse/CBL-1710).

**After enabling version-vector:**
* When calling `c4doc_resolveConflict`, always send `mergedFlags` so it will depend on LiteCore whether it wants to use the flags or not.
* Simplified testConflictResolutionDefault.
* Disabled failed tests per the open issues.
* Open Issue: [CBL-1713](https://issues.couchbase.com/browse/CBL-1713), [CBL-1715](https://issues.couchbase.com/browse/CBL-1715).

**Information about version vector and lite-core api changes by @snej**
* [LiteCore Version Vector Progress Report](https://docs.google.com/document/d/1OwHqwctXNd6bGqkhlPx8jWvHDnjdyLK3YU0iQHCRrWQ)
* [LiteCore Version Vector Progress Report #2](https://docs.google.com/document/d/1Wa5YaoPXgciGGVICY65JtiYhqhbeD64-tuptmliEGoA)
* [LiteCore Version Vector Progress Report #3](https://docs.google.com/document/d/153s2oD5qdtYGBDNYtYH6_c0wyrHiKVXF-23uNyA8lJA)
* [Information about c4db_getDoc and c4db_get](https://github.com/couchbase/couchbase-lite-core/blob/master/C/include/c4Document.h#L101-L122).

### CODE REVIEW

File: Objective-C/Internal/CBLC4Document.h [hunk 1]
```diff
@@ -26,13 +26,17 @@ NS_ASSUME_NONNULL_BEGIN
...
 @property (readonly, nonatomic) C4RevisionFlags revFlags;
 
+/** Sequence of the selected revision. */
 @property (readonly, nonatomic) C4SequenceNumber sequence;
 
+/** Revision ID of the selected revision. */
 @property (readonly, nonatomic) C4String revID;
 
-@property (readonly, nonatomic) C4Slice body;
+/** Body of the selected revision (Note this body ). */
```
Comment (pasin) [inline Objective-C/Internal/CBLC4Document.h:38]: Will remove `(Note this body )`.

File: Objective-C/CBLDatabase.mm [hunk 2]
```diff
@@ -1008,7 +1008,7 @@ - (BOOL) saveDocument: (CBLDocument*)document
                 
                 C4Error err;
                 CBLStringBytes bDocID(document.id);
-                curDoc = c4doc_get(_c4db, bDocID, true, &err);
+                curDoc = c4db_getDoc(_c4db, bDocID, true, kDocGetCurrentRev, &err);
```
Comment (borrrden) [inline Objective-C/CBLDatabase.mm:1011]: I searched through the three linked docs but couldn't find any reference to this function.  Could you point it out?

**Review by borrrden — APPROVED** [review]
I didn't evaluate the tests, but will address those separate as part of the ticket investigations

File: Objective-C/CBLDatabase.mm (hunk 2 above)
Comment (pasin) [reply Objective-C/CBLDatabase.mm:1011]: Apparently that function is not in the three version vector docs. The information can be found in the API doc itself. I will include this in the PR's description.

https://github.com/couchbase/couchbase-lite-core/blob/master/C/include/c4Document.h#L101-L122

**Review by jayahariv — APPROVED** [review]

**Review by Sandychuang8 — APPROVED** [review]

**Pull request merged**